lib/spack/spack/test/installer.py

```diff
@@ -252,6 +253,46 @@ def test_installer_str(install_mockery):
     assert "failed (0)" in istr
 
 
+def test_installer_prune_built_build_deps(install_mockery):
+    r"""
+    Ensure that build dependencies of installed deps are pruned
+    from installer package queues.
+
+                (a)
+               /   \
+             (d)   (b) <-- is installed already so we should
+              |      \     prune (c) from this install since
+             (e)     (c)   it is *only* needed to build (b)
+
+    Thus since (b) is already installed our build_pq dag should
+    only include four packages. [(a), (b), (d), (e)]
+    """
+    tmpdir = tempfile.mkdtemp()
```

Comment:
This has side effects, use the `tmpdir` fixture, see other tests

Reply:
I'd originally attempted to use the `tmpdir` fixture but it doesn't create an empty directory which is needed for the MockRepositoryBuilder. In other tests utilizing the MockRepositoryBuilder my assumption is that they use the explicit call to the `tempfile` library for this reason.

Reply:
Fixed! Let me know what you think of the new implementation. Sorry deleted my previous comment after digging into other tests more as well. Apologies for the confusion.